prog/hints.go

```diff
@@ -104,10 +104,18 @@ func generateHints(compMap CompMap, arg Arg, exec func()) {
 }
 
 func checkConstArg(arg *ConstArg, compMap CompMap, exec func()) {
-	original := arg.Val
+	argSize := arg.Size()
+	if argSize > 8 {
```

Comment:
The same comment as for the ints change applies here: if we got argSize>8 we are comparing kilograms with liters.